.github/scripts/format_dbt_test_failures.py

```diff
@@ -0,0 +1,256 @@
+    where each key is a name of a sheet and each associated value is
+    a list of failed tests for that sheet."""
+    conn = pyathena.connect(
+        s3_staging_dir=AWS_ATHENA_S3_STAGING_DIR,
+        region_name="us-east-1",
+        cursor_class=pyathena.cursor.DictCursor,
+    )
+    cursor = conn.cursor()
+
+    failed_tests_by_category: FailedTestsByCategory = {}
+
+    for result in run_results["results"]:
+        if result["status"] == "fail":
+            unique_id = result["unique_id"]
+            # Unique ID format is:
+            # test.athena.<test_name>.<hash>
+            test_name = unique_id.split(".")[2]
+
+            node = manifest["nodes"].get(unique_id)
+            if node is None:
+                raise ValueError(
+                    f"Missing dbt manifest node with id {unique_id}"
+                )
+
+            meta = node.get("meta", {})
+            category = meta.get("category", DEFAULT_TEST_CATEGORY)
+            test_description = meta.get("description")
+
+            # Search for the model that is implicated in this test via the
+            # depends_on key
+            models = node.get("depends_on", {}).get("nodes", [])
+            if not models:
+                raise ValueError(
+                    "Missing `depends_on.nodes` attribute for test "
+                    f"{test_name}"
+                )
+            # Cross-table comparisons often involve multiple models; we have
+            # determined experimentally that the second one is usually
+            # the model that the test is concerned with, although if this
+            # becomes a problem in the future we can update the test metadata
+            # to expose an override attribute
+            model = models[-1]
+
+            # Reference format for models/sources is:
+            # <"model"/"source">.athena.<schema_name>.<table_name>
+            table_name = model.split(".")[-1]
+
+            # Get the fully-qualified name of the table that stores failures
+            # for this test so that we can query it
+            test_results_relation_name = node.get("relation_name")
+            if test_results_relation_name is None:
+                raise ValueError(
+                    f"Missing relation_name attribute for test {test_name}. "
+                    "Did you run `dbt test` with the --store-failures flag?"
+                )
+
+            print(f"Querying failed rows from {test_results_relation_name}")
+            cursor.execute(f"select * from {test_results_relation_name}")
+            query_results = cursor.fetchall()
+            if len(query_results) == 0:
+                raise ValueError(
+                    f"Test {test_name} has status 'fail' but no failures"
+                )
+
+            # Add custom fields to query results that we don't expect to be
+            # included in the response
+            failed_tests = [
+                {
+                    "iasworld_table": table_name,
```

Comment:
**issue:** We also need to make this export compatible with the test _views_ tagged with `test_qc_inquire`. These are multi-table tests pulled from the Inquire system. I think the code already here should work for them, we just have to generalize the naming a bit, i.e.:

```suggestion
                    "source_table": table_name,
```